CRM/Sepa/Logic/Queue/Close.php

```diff
@@ -22,285 +22,402 @@
+    $status_in_progress = CRM_Sepa_Logic_Settings::contributionInProgressStatusId();
+
     if (empty($contribution_id_list)) {
-      // this would cause SQL errors
+      // This would cause SQL errors
       return;
     }
-    if ($this->target_status_id == $status_inProgress) {
-      // this status cannot be set via the API -> use SQL
-      CRM_Core_DAO::executeQuery("UPDATE civicrm_contribution SET contribution_status_id={$status_inProgress} WHERE id IN ({$contribution_id_list});");
 
-    } else { // this should be status 'Completed', but it doesn't really matter
-      // first, some sanity checks:
+    if ($target_status_id == $status_in_progress) {
+      // This status cannot be set via the API -> use SQL
+      CRM_Core_DAO::executeQuery("
+        UPDATE civicrm_contribution SET contribution_status_id = $status_in_progress
+        WHERE id IN ($contribution_id_list);
+      ");
+    } else {
+      // This should be status 'Completed', but it doesn't really matter
+      // Sanity checks:
       if (version_compare(CRM_Utils_System::version(), '4.7.0', '>=')) {
-        // make sure they're all in status 'In Progress' to avoid SEPA-514
-        CRM_Core_DAO::executeQuery("UPDATE civicrm_contribution SET contribution_status_id={$status_inProgress} WHERE id IN ({$contribution_id_list});");
+        // Make sure they're all in status 'In Progress' to avoid SEPA-514
+        CRM_Core_DAO::executeQuery("
+          UPDATE civicrm_contribution SET contribution_status_id = $status_in_progress
+          WHERE id IN ($contribution_id_list);
+        ");
       }
 
-      // then: set them all to the new status
+      // Set them all to the new status
       foreach ($contributions as $contribution) {
-        civicrm_api3('Contribution', 'create', array(
-            'id'                       => $contribution['id'],
-            'contribution_status_id'   => $this->target_status_id,
-            'receive_date'             => date('YmdHis', strtotime($this->txgroup['collection_date']))));
+        $receive_date = date('YmdHis', strtotime($txgroup['collection_date']));
+
+        civicrm_api3('Contribution', 'create', [
+          'contribution_status_id' => $target_status_id,
+          'id'                     => $contribution['id'],
+          'receive_date'           => $receive_date,
+        ]);
       }
     }
   }
+
+  /**
+   * Render a title for a task
+   *
+   * @param $mode string
+   * @param $params array
+   * @return string
+   */
+  private static function taskTitle($mode, $params = []) {
+    $counter = $params['counter'] ?? NULL;
+    $txgroup_ref = $params['txgroup_ref'] ?? NULL;
+
+    switch ($mode) {
+      case 'update_contribution':
+        return ts("Updating contributions in group '%1'... (%2)", [
+          1        => $txgroup_ref,
+          2        => $counter,
+          'domain' => 'org.project60.sepa',
+        ]);
+
+      case 'set_group_status':
+        return ts("Updating status of group '%1'", [
```

Comment:
Can we add `target_status_id` here to make debugging a bit easier (so we know which status we're setting)?